internal/clients/diffreviewer/circleci/circleci_service.go

```diff
@@ -0,0 +1,106 @@
+package circleci
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/nightfallai/nightfall_code_scanner/internal/clients/diffreviewer"
+	"github.com/nightfallai/nightfall_code_scanner/internal/clients/gitdiff"
+	"github.com/nightfallai/nightfall_code_scanner/internal/clients/logger"
+	circlelogger "github.com/nightfallai/nightfall_code_scanner/internal/clients/logger/circle_logger"
+	"github.com/nightfallai/nightfall_code_scanner/internal/clients/nightfall"
+	"github.com/nightfallai/nightfall_code_scanner/internal/interfaces/gitdiffintf"
+	"github.com/nightfallai/nightfall_code_scanner/internal/nightfallconfig"
+)
+
+const (
+	WorkspacePathEnvVar          = "GITHUB_WORKSPACE"
+	NightfallAPIKeyEnvVar        = "NIGHTFALL_API_KEY"
+	CircleRepoNameEnvVar         = "CIRCLE_PROJECT_REPONAME"
+	CircleOwnerNameEnvVar        = "CIRCLE_PROJECT_USERNAME"
+	CircleCurrentCommitShaEnvVar = "CIRCLE_SHA1"
+	CircleBeforeCommitEnvVar     = "EVENT_BEFORE"
+	CirclePullRequestUrlEnvVar   = "CIRCLE_PULL_REQUEST"
+)
+
+// Service contains the github client that makes Github api calls
+type Service struct {
+	Logger  logger.Logger
+	GitDiff gitdiffintf.GitDiff
+}
+
+// NewCircleCiService creates a new CircleCi service
+func NewCircleCiService() diffreviewer.DiffReviewer {
+	return &Service{
+		Logger: circlelogger.NewDefaultCircleLogger(),
+	}
+}
+
+// GetLogger gets the github service logger
+func (s *Service) GetLogger() logger.Logger {
+	return s.Logger
+}
+
+// LoadConfig gets all config values from files or environment and creates a config
+func (s *Service) LoadConfig(nightfallConfigFileName string) (*nightfallconfig.Config, error) {
+	s.Logger.Debug("Loading configuration")
+	workspacePath, ok := os.LookupEnv(WorkspacePathEnvVar)
+	if !ok {
+		s.Logger.Error(fmt.Sprintf("Environment variable %s cannot be found", WorkspacePathEnvVar))
+		return nil, errors.New("missing env var for workspace path")
+	}
+	commitSha, ok := os.LookupEnv(CircleCurrentCommitShaEnvVar)
```

Comment:
just checking to make sure - the `LookupEnv` `ok` return value is always true if the variable exists, regardless of whether it's empty. So if any of these Env vars cannot have valid empty values and are also not automatically set by circle, it's probably best to do an additional check to make sure the var isn't empty so we can make sure to log a helpful error message instead of just continuing on (e.g.  of `!ok && var != ""`)